src/utils/permissions.ts

```diff
@@ -79,6 +80,20 @@ export const isAppExecutor = async (d2: D2) => {
     return globalAdmin || !!userRoles.find((role: any) => role.name === name);
 };
 
+export const verifyUserHasAccessToSyncRule = async (d2: D2, syncRuleUId: string) => {
+    const appConfigurator = await isAppConfigurator(d2);
+    const globalAdmin = await isGlobalAdmin(d2);
+    const userInfo = await getUserInfo(d2);
+
+    if (globalAdmin) return true;
```

Comment:
If user is ``globalAdmin`` we can return before doing the other two async operations and it will "speed-up" a bit the check.

```
const globalAdmin = await isGlobalAdmin(d2);
if (globalAdmin) return true;
const appConfigurator = await isAppConfigurator(d2);
const userInfo = await getUserInfo(d2);
...
```

Reply:
Done!